### PR TITLE
[SH] Fix build warnings

### PR DESCRIPTION
**Pull request by Rot127**
These commits add several build warning options to `CMakeList.txt` and fixes build warnings for the SH arch.
Two unused functions are removed and a `parantheses`/`int-in-bool-contexxt` warning is fixed.

@ysat0 Could you please take a look at it as well?

### CODE REVIEW

**Comment by Rot127** [comment]
Anyone has an idea why `MSVC != true` for the failing test? Unfortunately, I currently don't have a Windows machine for debugging.

**Comment by ysat0** [comment]
I also don't have an environment where I can use MSVC right now.
I will prepare and check it in the next week.

**Comment by ysat0** [comment]
Fixed.
https://github.com/ysat0/capstone/commit/0eebf320ba047a398304ab6723009c8dcc5d6a34
It looks like cmake can't detect MSVC.
I've verified that this change builds on Windows/MSVC and Linux/gcc.

**Comment by Rot127** [comment]
The build seems to [fail with this as well](https://github.com/ysat0/capstone/actions/runs/4313653077/jobs/7525712015). Or am I missing something?
We could also just remove the build options and open an issue about it. If everyone is fine with it.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # For MSVC_RUNTIME_LIBRARY
 cmake_minimum_required(VERSION 3.15)
 
+if (MSVC)
```
Comment (kabeor) [inline CMakeLists.txt]: Hi, you need to define what `MSVC` is here

Comment (Rot127) [reply CMakeLists.txt]: It should be predefined by `cmake`. See: https://cmake.org/cmake/help/latest/variable/MSVC.html
But it isn't in our case.

Comment (Rot127) [reply CMakeLists.txt]: ~~Wait. I haven't tried [this](https://stackoverflow.com/questions/41692725/cmake-doesnt-recognize-msvc-compiler) last time.~~ Doesn't work

**Comment by XVilka** [comment]
@Rot127 I think I might know the reason - this variable probably doesn't work before the `project()` clause.

See also https://stackoverflow.com/questions/41692725/cmake-doesnt-recognize-msvc-compiler#41693234

**Comment by Rot127** [comment]
This was it. Thanks!

**Review by XVilka — APPROVED** [review]
LGTM

**Comment by kabeor** [comment]
Great, Thank you!

**Pull request merged**